### PR TITLE
Update packaging docs and other README maintenance

### PR DESCRIPTION
**Pull request by jordansissel**
Should address some of the "how to build packages" confusion in #284

I also removed some old text that wasn't relevant anymore.

### CODE REVIEW

**Comment by suyograo** [comment]
Can we add a note about not using `gccgo`

**Comment by jordansissel** [comment]
Added.

File: README.md [hunk 1]
```diff
@@ -128,17 +128,32 @@ Again, creating a correct SSL/TLS certificate authority or generally doing certi
...
 
+Note: Do not use gccgo for this project. If you don't know what that means,
+you're probably OK to ignore this.
+
         git clone git://github.com/elasticsearch/logstash-forwarder.git
         cd logstash-forwarder
         go build
 
+gccgo note: Using gccgo is not recommended because it produces a binary with a
+runtime dependency on libgo. With normal the normal go compiler, this
```
Comment (suyograo) [inline README.md]: 2 normal :)

**Comment by suyograo** [comment]
LGTM, minor typo

File: README.md (hunk 1 above)
Comment (jordansissel) [inline README.md]: Normally, with normal the normal go, we normally normalize the normals, normally. Welcome to normaltown.

**Pull request closed without merging**